doc/source/overview/high_level.md

```diff
@@ -0,0 +1,508 @@
+
+#### Accumulated Local Effects
+
+| Model-types | Task-types     | Data-types  |
+| ----------- | -------------- | ----------- |
+| Black-box   | Classification | Tabular     |
+|             | Regression     |             |
+
+
+Alibi only provides accumulated local effects plots because of the available global feature attribution methods they give the most accurate insight. Alternatives include Partial Dependence Plots. ALE plots work by averaging the local changes in a prediction at every instance in the data distribution. They then accumulate these differences to obtain a plot of prediction over the selected feature dependencies.
+
+Suppose we have a model $f$ and features $X=\{x_1,... x_n\}$. Given a subset of the features $X_S$, we denote $X_C=X \setminus X_S$. We want to obtain the ALE-plot for the features $X_S$, typically chosen to be at most a set of dimension two to be visualized easily. For simplicity assume we have $X=\{x_1, x_2\}$ and let $X_S=\{x_1\}$ so $X_C=\{x_2\}$. The ALE of $x_1$ is defined by:
+
+$$
+\hat{f}_{S, ALE}(x_1) =
+\int_{min(x_1)}^{x_1}\mathbb{E}\left[
+\frac{\partial f(X_1, X_2)}{\partial X_1} | X_1 = z_1
+\right]dz_1 - c_1
+$$
+
+The term within the integral computes the expectation of the model derivative in $x_1$ over the random variable $X_2$ conditional on $X_1=z_1$. By taking the expectation for $X_2$, we factor out its dependency. So now we know how the prediction $f$ changes local to a point $X_1=z_1$ independent of $X_2$. Integrating these changes over $x_1$ from a minimum value to the value of interest, we obtain the global plot of how the model depends on $x_1$. ALE-plots get their names as they accumulate (integrate) the local effects (the expected partial derivatives). Note that here we have assumed $f$ is differentiable. In practice, however, we compute the various quantities above numerically, so this isn't a requirement.
+
+__TODO__:
+- Add picture explaining the above idea.
+
+For more details on accumulated local effects including a discussion on PDP-plots and M-plots see
+[Motivation-and-definition for ALE](../methods/ALE.ipynb#Motivation-and-definition)
+
+:::{admonition} **Note 4: Categorical Variables and ALE**
+Note that because ALE plots require computing differences between variables, they don't naturally extend to categorical data unless there is a sensible ordering on the data. As an example, consider the months of the year. To be clear, this is only an issue if the variable you are taking the ALE for is categorical.
+:::
+
+**Pros**:
+- ALE-plots are easy to visualize and understand intuitively
+- Very general as it is a black box algorithm
+- Doesn't struggle with dependencies in the underlying features, unlike PDP plots
+- ALE plots are fast
+
+**Cons**:
+- Harder to explain the underlying motivation behind the method than PDP plots or M plots.
+- Requires access to the training dataset.
+- Unlike PDP plots, ALE plots do not work with Categorical data
+
+### Counter Factuals:
```

Comment:
Counter Factuals --> Counterfactual instances (globally)

What's worked best is to introduce the concept as "Counterfactual instances/examples" and thereafter refer to them as "Counterfactuals".